swarm/storage/netstore_test.go

```diff
@@ -1,702 +0,0 @@
-// Copyright 2018 The go-ethereum Authors
```

Comment:
Really all of netstore_test must go? Is there nothing to be unit-tested for Netstore anymore?

Reply:
`NetStore`'s internal impl. has changed, so there particular tests are redundant. If you check them out, you will see that they were impl. specific.

Regarding test coverage for the `NetStore`, it is being used by all simulations and by the smoke tests. This gives us certain confidence that it is behaving correctly.

What other tests would you like to see here? Could you describe some of them in detail?

---

For now I focus on the outstanding TODOS, which might change the `NetStore`, and once we have those sorted out, we can think about locking the impl. with unit tests if necessary.

Reply:
Potential tests :
 * Write chunk via `NetStore.Put` and check that it is there
 * Write chunk via `NetStore.Put` check no change to existing fetchers
 * Write chunk via `NetStore.Put`, create a dummy fetcher and check that it got terminated
 * PR desc says: "NetStore.Put is responsible for removing fetchers and closing the delivered channel." - close `NetStore` and check that fetchers are removed and delivered channel closed


etc.
Every exported function should have a unit test I believe

Reply:
This code path is hit multiple times by integration tests. If NetStore doesn't work correctly, many of our existing unit tests and integration tests will fail - such as Swarm Feeds, etc.

I see very little value in adding mocks, dummy fetchers, and more code that locks this impl. - we might actually want to refactor it once we come up with a better structure for the syncing/stream protocol and other consumers.